metaflow/plugins/kfp/kfp.py

```diff
@@ -231,15 +231,23 @@ def create_workflow_yaml(self, output_path: str, output_format: str = "kfp") ->
         if output_format == "kfp":
             pass  # KFP pipeline yaml is created by default
         elif output_format == "argo-workflow":
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
+            # Output of KFP compiler already has workflow["kind"] = "Workflow".
+
+            # Keep generateName - Argo Workflow is usually used in single run.
+
+            # Service account is added through webhooks.
+            workflow["spec"].pop("serviceAccountName", None)
```

Comment:
Are you going to open a webhook PR to add in the service account to workflow pods?

Reply:
there is no additional work to distinguish the pods as we are already targeting Argo created pods with this [label](https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/aip-admission-webhook-service/-/blob/main/aip_admission_webhook_service/config.py#L8). your change should be a quick one liner. I think it would be worthwhile to move this config now.

Reply:
Cool. Working on it now.

Reply:
https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/aip-admission-webhook-service/-/merge_requests/19

Reply:
MR created: [aip-admission-webhook-service!19](https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/aip-admission-webhook-service/-/merge_requests/19)